### PR TITLE
OAuth tests

### PR DESCRIPTION
**Pull request by OmeGak**

### CODE REVIEW

**Comment by ThiefMaster** [comment]
Did you simply replace `dummy_user` with `dummy_user_legacy`? I think in the majority of the tests you don't need it because we were already getting the new one from it.

For the other cases, `dummy_user.as_avatar` would probably work fine instead of adding a new fixture...

File: indico/modules/oauth/models/tokens_test.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    assert OAuthGrant.make_key(client_id, code) == key
+
+
+def test_grant_delete(mocker, dummy_grant):
+    mocker.patch.object(OAuthGrant, '_cache')
+    dummy_grant.delete()
+    OAuthGrant._cache.delete.assert_called_with(dummy_grant.key)
+
+
+@freeze_time(datetime.utcnow())
```
Comment (ThiefMaster) [inline indico/modules/oauth/models/tokens_test.py]: Don't we have a fixture for that?

File: indico/modules/oauth/provider_test.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    assert grant.code == code['code']
+    assert grant.redirect_uri == request.redirect_uri
+    assert grant.user == session.user
+    assert grant.scopes == request.scopes
+    assert grant.expires == expires
+    assert grant.save.called
+
+
+@pytest.mark.usefixtures('request_context')
+@pytest.mark.parametrize(('access_token'), (True, False))
```
Comment (ThiefMaster) [inline indico/modules/oauth/provider_test.py]: the `()` around `'access_token'` make no sense. If you wanted a tuple, you'd need a `,` in there... but pytest is smart enough to let you specify a string if you have only one element - so you can simply remove them.

File: indico/modules/oauth/testing/fixtures.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+from indico.modules.oauth.models.applications import OAuthApplication
+from indico.modules.oauth.models.tokens import OAuthToken
+
+
+@pytest.fixture
+def create_application(db, dummy_event, dummy_registrant):
+    """Returns a callable which lets you create applications"""
+
+    def _create_application(name, **params):
+        params.setdefault('client_id', str(uuid4()))
```
Comment (ThiefMaster) [inline indico/modules/oauth/testing/fixtures.py]: not `unicode`?

File: indico/modules/oauth/models/tokens_test.py (hunk 1 above)
Comment (OmeGak) [inline indico/modules/oauth/models/tokens_test.py]: True, I just realized, but what is the drawback of using the decorator?

File: indico/modules/oauth/provider_test.py (hunk 2 above)
Comment (OmeGak) [inline indico/modules/oauth/provider_test.py]: I know, thanks. I had more elements before and I simply forgot to remove the parenthesis.

File: indico/modules/oauth/models/tokens_test.py (hunk 1 above)
Comment (ThiefMaster) [inline indico/modules/oauth/models/tokens_test.py]: Doesn't integrate that well with pytest's magic in some cases

Comment (OmeGak) [inline indico/modules/oauth/models/tokens_test.py]: Always the fixture, then? In `agreements_test.py` we used the decorator.

**Comment by OmeGak** [comment]
All cool about `dummy_avatar` returning `dummy_user.as_avatar` except that in `rb` tests, we are, for some reason, using both `dummy_avatar.user` and `dummy_user` as a cheap way to have two different users. I'll change those tests to use `create_user` instead and then implement the suggestion.

**Comment by ThiefMaster** [comment]
I've adapted all the RB code to work with Users a few weeks ago... so it shouldn't need legacy avatars.

Also, can you show me a test where we use two dummy users as different users? That sounds *very* strange, especially considering that `some_avatar` and `some_avatar.user` actually compare as equal...

**Comment by OmeGak** [comment]
Nvm, it was due to some replacement I did in a previous commit. Fixed everything already, I believe. Green light to merge? :vertical_traffic_light:

**Comment by ThiefMaster** [comment]
I didn't have a closer look yet but if you squash those fixups I could do so once I'm back (I guess it's not super urgent anyway) unless someone else feels like doing it in the meantime

**Comment by pferreir** [comment]
I also had a quick look and it looks fine to me. Really nothing to add.

**Pull request merged**